resource_control: ru consistent with serverless (#14081)
pingcap/docs: tidb-resource-control.md

<table>
    <thead>
        <tr>
            <th>Resource Type</th>
            <th>RU Consumption</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan="3">Read</td>
            <td>2 storage read batches consume 1 RU</td>
        </tr>
        <tr>
            <td>8 storage read requests consume 1 RU</td>
        </tr>
            <td>64 KiB read request payload consumes 1 RU</td>
        <tr>
            <td rowspan="3">Write</td>
            <td>1 storage write batch consumes 1 RU * (the number of replicas)</td>
        </tr>
        <tr>
            <td>1 storage write request consumes 1 RU</td>
        </tr>
        <tr>
            <td>1 KiB write request payload consumes 1 RU</td>
        </tr>
        <tr>
            <td>SQL CPU</td>
            <td> 3 ms consumes 1 RU</td>
        </tr>
    </tbody>
</table>

> **Note:**
>
> - Each write operation is eventually replicated to all replicas (by default TiKV has 3 replicas). Each replication operation is considered a different write operation.
> - In addition to queries executed by the user, RU can be consumed by background tasks, such as automated statistical information collection.

> > - The preceding table lists only the resources involved in computing RUs for on-premises TiDB clusters, excluding the network and storage components. For TiDB Serverless RUs, see [TiDB Serverless Pricing Details](https://www.pingcap.com/tidb-cloud- serverless-pricing-details/).

Reviewer: ```suggestion
> - The preceding table lists only the resources involved in computing RUs for on-premises TiDB clusters, excluding the network and storage components. For TiDB Serverless RUs, see [TiDB Serverless Pricing Details](https://www.pingcap.com/tidb-cloud-serverless-pricing-details/).
```

## What is Request Unit (RU)

> Request Unit (RU) is a unified abstraction unit in TiDB for system resources, which currently includes CPU, IOPS, and IO bandwidth metrics. It is used to indicate the amount of resources consumed by a single request to the database. The number of RUs consumed by a request depends on a variety of factors, such as the type of operation, and the amount of data being retrieved or modified. Currently, the RU contains statistics for the resources in the following table:

Reviewer 2: ```suggestion
Request Unit (RU) is a unified abstraction unit in TiDB for system resources, which currently includes CPU, IOPS, and IO bandwidth metrics. It is used to indicate the amount of resources consumed by a single request to the database. The number of RUs consumed by a request depends on a variety of factors, such as the type of operations, and the amount of data being queried or modified. Currently, the RU contains consumption statistics for the resources in the following table:
```

Reviewer 2: reason: using "consumption statistics" to differentiate it from https://docs.pingcap.com/tidb/dev/statistics

<table>
    <thead>
        <tr>

>             <th>Resource Type</th>

Reviewer 2: ```suggestion
            <th>Resource type</th>
```

>             <th>RU Consumption</th>

Reviewer 2: ```suggestion
            <th>RU consumption</th>
```

        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan="3">Read</td>
            <td>2 storage read batches consume 1 RU</td>
        </tr>
        <tr>
            <td>8 storage read requests consume 1 RU</td>
        </tr>
        <tr>
            <td>64 KiB read request payload consumes 1 RU</td>
        </tr>
        <tr>
            <td rowspan="3">Write</td>

>             <td>1 storage write batch consumes 1 RU * (the number of replicas)</td>

Reviewer 2: ```suggestion
            <td>1 storage write batch consumes 1 RU for each replica</td>
```

        </tr>
        <tr>
            <td>1 storage write request consumes 1 RU</td>
        </tr>
        <tr>
            <td>1 KiB write request payload consumes 1 RU</td>
        </tr>
        <tr>
            <td>SQL CPU</td>
            <td> 3 ms consumes 1 RU</td>
        </tr>
    </tbody>
</table>

> **Note:**
>

> > - Each write operation is eventually replicated to all replicas (by default TiKV has 3 replicas). Each replication operation is considered a different write operation.

Reviewer 2: ```suggestion
> - Each write operation is eventually replicated to all replicas (by default, TiKV has 3 replicas). Each replication operation is considered a different write operation.
```

> > - In addition to queries executed by the user, RU can be consumed by background tasks, such as automated statistical information collection.

Reviewer 2: ```suggestion
> - In addition to queries executed by users, RU can be consumed by background tasks, such as automatic statistics collection.
```

> > - The preceding table lists only the resources involved in computing RUs for on-premises TiDB clusters, excluding the network and storage components. For TiDB Serverless RUs, see [TiDB Serverless Pricing Details](https://www.pingcap.com/tidb-cloud-serverless-pricing-details/).

Reviewer 2: ```suggestion
> - The preceding table lists only the resources involved in RU calculation for TiDB Self-Hosted clusters, excluding the network and storage consumption. For TiDB Serverless RUs, see [TiDB Serverless Pricing Details](https://www.pingcap.com/tidb-cloud-serverless-pricing-details/).
```
